modules/base/rendering/renderabledisc.cpp

```diff
@@ -0,0 +1,230 @@
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/openglstatecache.h>
+#include <ghoul/opengl/programobject.h>
+#include <ghoul/opengl/textureunit.h>
+
+namespace {
+    constexpr const char* _loggerCat = "RenderableDisc";
+
+    constexpr const std::array<const char*, 4> UniformNames = {
+        "modelViewProjectionTransform", "opacity", "width", "colorTexture"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo TextureInfo = {
+        "Texture",
+        "Texture",
+        "This value is the path to a texture on disk that contains a one-dimensional "
+        "texture to be used for the color."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SizeInfo = {
+        "Size",
+        "Size",
+        "This value specifies the outer radius of the disc in meter."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo WidthInfo = {
+        "Width",
+        "Width",
+        "This value is used to set the width of the disc. The actual width is set "
+        "based on the given size and this value should be set between 0 and 1. A value "
+        "of 1 results in a full circle and 0.5 a disc with an inner radius of 0.5*size."
+    };
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableDisc::Documentation() {
+    using namespace documentation;
+    return {
+        "Renderable Disc",
+        "renderable_disc",
+        {
+            {
+                TextureInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                TextureInfo.description
+            },
+            {
+                SizeInfo.identifier,
+                new DoubleVerifier,
+                Optional::Yes,
+                SizeInfo.description
+            },
+            {
+                WidthInfo.identifier,
+                new DoubleVerifier,
+                Optional::Yes,
+                WidthInfo.description
+            }
+        }
+    };
+}
+
+RenderableDisc::RenderableDisc(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _texturePath(TextureInfo)
+    , _size(SizeInfo, 1.f, 0.f, 1e13f)
+    , _width(WidthInfo, 0.5f, 0.f, 1.f)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableDisc"
+    );
+
+    _texturePath = absPath(dictionary.value<std::string>(TextureInfo.identifier));
+    _texturePath.onChange([&]() { _texture->loadFromFile(_texturePath); });
+    addProperty(_texturePath);
+
+    if (dictionary.hasKey(SizeInfo.identifier)) {
+        _size = static_cast<float>(dictionary.value<double>(SizeInfo.identifier));
+    }
+    setBoundingSphere(_size);
+    _size.onChange([&]() { _planeIsDirty = true; });
+    addProperty(_size);
+
+    if (dictionary.hasKey(WidthInfo.identifier)) {
+        _width = static_cast<float>(dictionary.value<double>(WidthInfo.identifier));
+    }
+    addProperty(_width);
+    addProperty(_opacity);
```

Comment:
feels like opacity should be read from the dictionary also

Reply:
That is done in the constructor of the base class ( `Renderable` ) :)

Reply:
Then I vote that the `addProperty(_opacity);` goes to the base class for consistency.